[AArch64] Cleanup existing values in getMemOpInfo

davemgreen opened the pull request:
This patch tries to clean up some of the existing values in getMemOpInfo. All values should now be in bytes (not bits), and the MinOffset/MaxOffset are now always represented unscaled (the immediate that will be present in the final instruction).

Although I could not find a place where it altered codegen, the offset of a post-index instruction will be 0, not scale*imm. A IsPostIndexLdStOpcode method has been added to try and make sure that case is handled properly.

davemgreen commented on llvm/lib/Target/AArch64/AArch64InstrInfo.cpp:
```diff
@@ -3731,17 +3954,17 @@ bool AArch64InstrInfo::getMemOpInfo(unsigned Opcode, TypeSize &Scale,
   case AArch64::STPDpre:
   case AArch64::LDPDpost:
     Scale = TypeSize::getFixed(8);
     Width = TypeSize::getFixed(8);
```
Yeah that sounds good, I forgot to change that one. Anything else you think should change?

romainthomas replied:
Nop looks good :+1:

efriedma-quic reviewed (approved):
LGTM

It's a little surprising how broken this was without anyone noticing...

sdesmalen-arm commented on llvm/lib/Target/AArch64/AArch64InstrInfo.cpp:
```diff
@@ -3722,26 +3945,26 @@ bool AArch64InstrInfo::getMemOpInfo(unsigned Opcode, TypeSize &Scale,
-    Width = TypeSize::getFixed(16);
-    MinOffset = -1024;
-    MaxOffset = 1008;
+    Width = TypeSize::getFixed(16 * 2);
```
nit: these case statement can be combined with the ones for `LDPQi`, etc?

sdesmalen-arm commented on llvm/lib/Target/AArch64/AArch64InstrInfo.cpp:
```diff
@@ -3722,26 +3945,26 @@ bool AArch64InstrInfo::getMemOpInfo(unsigned Opcode, TypeSize &Scale,
-    Width = TypeSize::getFixed(8);
-    MinOffset = -512;
-    MaxOffset = 504;
+    Width = TypeSize::getFixed(8 * 2);
```
nit: Same for these, but then wrt `LDPXi`

sdesmalen-arm reviewed (approved):
Two little nits, but otherwise LGTM

davemgreen committed:
[AArch64] Cleanup the values in getMemOpInfo

This patch tries to clean up some of the existing values in getMemOpInfo. All
values should now be in bytes (not bits), and the MinOffset/MaxOffset are now
always represented unscaled (the immediate that will be present in the final
instruction).

Although I could not find a place where it altered codegen, the offset of a
post-index instruction will be 0, not scale*imm. A IsPostIndexLdStOpcode method
has been added to try and make sure that case is handled properly.
